Examples/Scripts/MaterialMapping/material_comparison.py

```diff
@@ -0,0 +1,239 @@
+        tprof.GetName() + "_th1",
+        tprof.GetTitle(),
+        tprof.GetNbinsX(),
+        tprof.GetXaxis().GetXmin(),
+        tprof.GetXaxis().GetXmax(),
+    )
+    for i in range(tprof.GetNbinsX()):
+        if tprof.GetBinContent(i + 1) == 0.0:
+            h.SetBinContent(i + 1, 0.0)
+            h.SetBinError(i + 1, 1000.0)
+            continue
+        h.SetBinContent(i + 1, tprof.GetBinContent(i + 1))
+        h.SetBinError(i + 1, tprof.GetBinError(i + 1))
+    return h
+
+
+if "__main__" == __name__:
+    p = argparse.ArgumentParser()
+
+    p.add_argument(
+        "-e", "--entries", type=int, default=100000, help="Number of events to process"
+    )
+    p.add_argument(
+        "-i",
+        "--input",
+        type=str,
+        nargs="+",
+        default="",
+        help="Input files with material tracks",
+    )
+    p.add_argument(
+        "-o",
+        "--output",
+        type=str,
+        default="",
+        help="Output file with produced material overview plots",
+    )
+    p.add_argument(
+        "-l",
+        "--labels",
+        type=str,
+        nargs="+",
+        default="",
+        help="The labels for the input files",
+    )
+    p.add_argument(
+        "-v",
+        "--variables",
+        type=str,
+        nargs="+",
+        default=["t_X0", "t_L0"],
+        help="The variables to be plotted",
+    )
+    p.add_argument(
+        "-m",
+        "--max",
+        type=float,
+        nargs="+",
+        default=[4.5, 1.5],
+        help="The variables to be plotted",
+    )
+    p.add_argument(
+        "-a",
+        "--axes",
+        type=str,
+        nargs="+",
+        default=["v_eta", "v_phi"],
+        help="The axes versus which the variables to be plotted",
+    )
+    p.add_argument(
+        "--eta",
+        type=float,
+        nargs=2,
+        default=[-3.0, 3.0],
+        help="Eta range for the plotting",
+    )
+    p.add_argument("--eta-bins", type=int, default=60, help="Eta bins for the plotting")
+    p.add_argument(
+        "--phi",
+        type=float,
+        nargs=2,
+        default=[-math.pi, math.pi],
+        help="Phi range for the plotting",
+    )
+    p.add_argument("--phi-bins", type=int, default=72, help="Phi bins for the plotting")
+    p.add_argument(
+        "-d",
+        "--detray-input",
+        type=str,
+        default="",
+        help="Optional detray input csv file",
+    )
+
+    args = p.parse_args()
+
+    ttrees = []
+
+    if len(args.input) != len(args.labels):
+        print("** ERROR ** The number of input files and labels must match")
+        exit(1)
+
+    # The histogram map
+    h_dict = {}
+    tfiles = []
+
+    ofile = ROOT.TFile.Open(args.output, "RECREATE") if args.output else None
+
+    input_files = args.input
+
+    # Detray csv file
+    if args.detray_input:
+        input_files.append(args.detray_input)
+
+    # Loop over the files and create the comparison histograms
+    for fi, ifile in enumerate(args.input):
```

Comment:
Here shouldn't you be looping over input_files instead ?
(it might do the same but it would be a bit cleaner I think)

Reply:
You are right, I will change.